infrastructure/kube/keep-test/btcdiff/btc-difficulty-statefulset.yaml

```
@@ -0,0 +1,79 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: btc-difficulty-maintainer
+  labels:
+    app: maintainer
+    type: btc-difficulty
+    network: goerli
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: maintainer
+      type: btc-difficulty
+      network: goerli
+  serviceName: btc-difficulty-maintainer
+  template:
+    metadata:
+      labels:
+        app: maintainer
+        type: btc-difficulty
+        network: goerli
+    spec:
+      volumes:
+      - name: eth-account-keyfile
+        configMap:
+          name: btc-difficulty-maintainer-info
+          items:
+          - key: btc-difficulty-maintainer-keyfile
+            path: btc-difficulty-maintainer-keyfile
+      - name: btc-difficulty-maintainer-config
+        configMap:
+          name: btc-difficulty-maintainer-config
+          items:
+          - key: btc-difficulty-maintainer-config.toml
+            path: btc-difficulty-maintainer-config.toml
+      containers:
+        - name: btc-difficulty-maintainer
+          image: gcr.io/keep-test-f3e0/keep-client:latest
+          imagePullPolicy: Always
+          resources:
+            requests:
+              cpu: 1500m
+              memory: 512M
+          env:
+          - name: LOG_LEVEL
+            value: info
+          - name: KEEP_ETHEREUM_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: btc-difficulty-maintainer-eth-password
+                key: eth-account-password
+          - name: ETH_WS_URL
+            valueFrom:
+              secretKeyRef:
+                name: eth-network-goerli
+                key: ws-url
+          volumeMounts:
+          - name: eth-account-keyfile
+            mountPath: /mnt/btc-difficulty-maintainer/keyfile
+          - name: btc-difficulty-maintainer-config
+            mountPath: /mnt/btc-difficulty-maintainer/config
+          command:
+          - keep-client
+          - maintainer
+          args:
+          - --bitcoinDifficulty
+          # run as developer to be able to set LightRelay address
+          - --developer
+          - --ethereum.url
+          - $(ETH_WS_URL)
+          - --ethereum.keyFile
+          - /mnt/btc-difficulty-maintainer/keyfile/btc-difficulty-maintainer-keyfile
+          - --bitcoin.electrum.url
+          - "electrum.blockstream.info:50001"
```

We could read the value from a config map for consistency.

In `kustomization.yaml` add:
```yaml
configMapGenerator:
  - name: electrum-api-mainnet
    literals:
      - electrumx-url-tcp=tcp://electrum.blockstream.info:50001
```

In the stateful set definition add:
```yaml
          env:
            - name: ELECTRUM_API_URL
              valueFrom:
                configMapKeyRef:
                  name: electrum-api-testnet
                  key: electrumx-url-wss
```

And update this line:
```suggestion
          - $(ELECTRUM_API_URL)
```


Also, we need to make sure the client parses the URL `tcp://electrum.blockstream.info:50001` correctly. It's common in our other apps to parse the protocol, host, and port out from such URL.

---

Done in 91481b3c88af9a877aa9d8d17eada4a7948056c6.
`tcp://electrum.blockstream.info:50001` did not work, but `electrum.blockstream.info:50001` worked correctly.

---

I added two issues to the backlog:
https://github.com/keep-network/keep-core/issues/3518
https://github.com/keep-network/keep-core/issues/3519